bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerMetadata.java

```diff
@@ -486,89 +482,94 @@ public static LedgerMetadata parseConfig(byte[] bytes, Optional<Long> msCtime) t
+
         // remaining size is total minus the length of the version line and '\n'
         char[] configBuffer = new char[config.length() - (versionLine.length() + 1)];
         if (configBuffer.length != reader.read(configBuffer, 0, configBuffer.length)) {
             throw new IOException("Invalid metadata buffer");
         }
 
-        LedgerMetadataFormat.Builder builder = LedgerMetadataFormat.newBuilder();
+        LedgerMetadataFormat.Builder formatBuilder = LedgerMetadataFormat.newBuilder();
+        TextFormat.merge((CharSequence) CharBuffer.wrap(configBuffer), formatBuilder);
+        LedgerMetadataFormat data = formatBuilder.build();
 
-        TextFormat.merge((CharSequence) CharBuffer.wrap(configBuffer), builder);
-        LedgerMetadataFormat data = builder.build();
-        lc.writeQuorumSize = data.getQuorumSize();
-        if (data.hasCtime()) {
-            lc.ctime = data.getCtime();
-            lc.storeSystemtimeAsLedgerCreationTime = true;
-        } else if (msCtime.isPresent()) {
-            lc.ctime = msCtime.get();
-            lc.storeSystemtimeAsLedgerCreationTime = false;
-        }
+        builder.withEnsembleSize(data.getEnsembleSize());
+        builder.withWriteQuorumSize(data.getQuorumSize());
         if (data.hasAckQuorumSize()) {
-            lc.ackQuorumSize = data.getAckQuorumSize();
+            builder.withAckQuorumSize(data.getAckQuorumSize());
+        } else {
+            builder.withAckQuorumSize(data.getQuorumSize());
+        }
+
+        if (data.hasCtime()) {
+            builder.withCreationTime(data.getCtime());
         } else {
-            lc.ackQuorumSize = lc.writeQuorumSize;
+            metadataStoreCtime.ifPresent(t -> builder.withMetadataStoreCreationTime(t));
         }
 
-        lc.ensembleSize = data.getEnsembleSize();
-        lc.length = data.getLength();
-        lc.state = data.getState();
-        lc.lastEntryId = data.getLastEntryId();
+        if (data.getState() == LedgerMetadataFormat.State.IN_RECOVERY) {
+            builder.withInRecoveryState();
+        } else if (data.getState() == LedgerMetadataFormat.State.CLOSED) {
+            builder.closingAt(data.getLastEntryId(), data.getLength());
+        }
 
         if (data.hasPassword()) {
-            lc.digestType = data.getDigestType();
-            lc.password = data.getPassword().toByteArray();
-            lc.hasPassword = true;
+            builder.withPassword(data.getPassword().toByteArray())
+                .withDigestType(protoToApiDigestType(data.getDigestType()));
         }
 
         for (LedgerMetadataFormat.Segment s : data.getSegmentList()) {
-            ArrayList<BookieSocketAddress> addrs = new ArrayList<BookieSocketAddress>();
-            for (String member : s.getEnsembleMemberList()) {
-                addrs.add(new BookieSocketAddress(member));
+            List<BookieSocketAddress> addrs = new ArrayList<>();
+            for (String a : s.getEnsembleMemberList()) {
```

Comment:
using a variable 'a' is generally not a good practice. why are you changing 'member' to 'a'?

Reply:
changed it to stream() implementation, compiled and it complained about BookieSocketAddress exceptions so changed it back, but kept variable names. Will revert.